src/Appwrite/Platform/Workers/Audits.php

```diff
@@ -131,21 +131,22 @@ public function action(Message $message, callable $getProjectDB, Document $proje
             return new NoCommit();
         }
 
-        try {
-            foreach ($this->logs as $sequence => $projectLogs) {
-                $dbForProject = $getProjectDB($projectLogs['project']);
-
-                Console::log('Processing batch with ' . count($projectLogs['logs']) . ' events');
+        $logs = $this->logs;
+        $this->logs = [];
+
+        foreach ($logs as $projectLogs) {
+            try {
+                $projectDocument = $projectLogs['project'];
+                Console::log('Processing Project "' . $projectDocument->getSequence() . '" batch with ' . count($projectLogs['logs']) . ' events');
+                $dbForProject = $getProjectDB($projectDocument);
                 $audit = new Audit($dbForProject);
-
                 $audit->logBatch($projectLogs['logs']);
                 Console::success('Audit logs processed successfully');
-
-                unset($this->logs[$sequence]);
+            } catch (Throwable $e) {
+                Console::error('Error processing audit logs for Project "' . $projectDocument->getSequence() . '": ' . $e->getMessage());
             }
```

Comment:
Instead of resetting all, let's unset at the end of the loop as before, but also unset in the catch